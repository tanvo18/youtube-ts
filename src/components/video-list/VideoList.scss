.video-list {
  padding: 0;
  list-style-type: none;

  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding: 0 15px;

  & li {
    width: 25%;
    padding-bottom: 20px;
  }
}
