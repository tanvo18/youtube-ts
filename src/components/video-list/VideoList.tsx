import React, { useState, useEffect } from 'react';
import VideoItem from '../video-item/VideoItem';
import * as videoAPI from '../../api/videoAPI';
import './VideoList.scss';
import { roundNumber, calculateDate } from '../../utils/common';

const VideoList: React.FC = (): JSX.Element => {
  const handleClickVideo = (event: MouseEvent, videoId: string) => {
    
  }

  const [videos, setVideos] = useState([]);

  useEffect(() => {
    videoAPI.getPopularVideos()
    .then(response => {
      // Normalize data
      let normalizeVideos = response.data.items;
      normalizeVideos.forEach((videoItem: any) => {
        videoItem.statistics.viewCount = roundNumber(videoItem.statistics.viewCount)
        videoItem.snippet.publishedAt = calculateDate(videoItem.snippet.publishedAt);
      })
      setVideos(normalizeVideos);
    })
  }, []);

  return (
    <div className="container video-container">
      <div>
        <ul className="video-list">
          {videos && videos.length > 0 ?
            videos.map((videoItem, index) => (
              <VideoItem
                key={index}
                video={videoItem}
                onClickVideo={handleClickVideo}
              />
            ))
            : <div>There are no videos</div>
          }
        </ul>
      </div>
    </div>
  )
}

export default VideoList
