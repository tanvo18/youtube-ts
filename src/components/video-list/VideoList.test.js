import React from 'react';
import VideoList from './VideoList';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

const props = {
  videos: [
    {
      'kind': 'youtube#video',
      'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/bImmdKn-oBhREMLFIIIiIKcWaT0"',
      'id': 'cBVGlBWQzuc',
      'snippet': {
        'publishedAt': 'a day ago',
        'channelId': 'UCN1hnUccO4FD5WfM7ithXaw',
        'title': 'Maroon 5 - Girls Like You ft. Cardi B (Volume 2)',
        'description': '"Girls Like You” is out now.\nhttp://smarturl.it/GLY \n\nFor more, visit: \nhttps://www.facebook.com/maroon5 \nhttps://twitter.com/maroon5 \nhttps://www.instagram.com/maroon5 \n\nSign up for updates: http://smarturl.it/Maroon5.News\n\nMusic video by Maroon 5 performing Girls Like You. © 2018 Interscope Records\n\nhttp://vevo.ly/BGhT8W',
        'thumbnails': {
          'default': {
            'url': 'https://i.ytimg.com/vi/cBVGlBWQzuc/default.jpg',
            'width': 120,
            'height': 90
          },
          'medium': {
            'url': 'https://i.ytimg.com/vi/cBVGlBWQzuc/mqdefault.jpg',
            'width': 320,
            'height': 180
          },
        },
        'channelTitle': 'Maroon5VEVO',
      },
      'statistics': {
        'viewCount': '1M',
        'likeCount': '128K',
        'dislikeCount': '1K',
        'favoriteCount': '0',
        'commentCount': '10027'
      }
    }
  ],
  statisticData: [
    {
      'kind': 'youtube#video',
      'etag': '"XI7nbFXulYBIpL0ayR_gDh3eu1k/QKYIcBPd8bwSw1iqefGNRk4ZkeY"',
      'id': 'cBVGlBWQzuc',
      'snippet': {
        'publishedAt': 'a day ago',
        'channelId': 'UCN1hnUccO4FD5WfM7ithXaw',
        'title': 'Maroon 5 - Girls Like You ft. Cardi B (Volume 2)',
        'description': '"Girls Like You” is out now.\nhttp://smarturl.it/GLY \n\nFor more, visit: \nhttps://www.facebook.com/maroon5 \nhttps://twitter.com/maroon5 \nhttps://www.instagram.com/maroon5 \n\nSign up for updates: http://smarturl.it/Maroon5.News\n\nMusic video by Maroon 5 performing Girls Like You. © 2018 Interscope Records\n\nhttp://vevo.ly/BGhT8W',
        'thumbnails': {
          'default': {
            'url': 'https://i.ytimg.com/vi/cBVGlBWQzuc/default.jpg',
            'width': 120,
            'height': 90
          },

        },
        'channelTitle': 'Maroon5VEVO',
        'categoryId': '10',
      },

      'statistics': {
        'viewCount': '1M',
        'likeCount': '128K',
        'dislikeCount': '1K',
        'favoriteCount': '0',
        'commentCount': '10027'
      }
    },
  ]
};

const emptyVideos = {
  videos: []
};

describe('test VideoList', () => {
  it('should render VideoItem', () => {
    const wrapper = shallow(<VideoList {...props} />);
    expect(wrapper.find('VideoItem').length).toBeGreaterThan(0);
  });

  it('should not render VideoItem with empty props', () => {
    const wrapper = mount(<VideoList {...emptyVideos} />);
    console.log('wrapper ', wrapper.find('.video-list div').debug());
    expect(wrapper.find('.video-list div').text()).toBe('There are no videos');
  });

  it('should render correctly', () => {
    const tree = renderer
      .create(<Router><VideoList {...props} /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
