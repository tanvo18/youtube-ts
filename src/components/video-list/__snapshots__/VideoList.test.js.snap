// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test VideoList should render correctly 1`] = `
<div
  className="container video-container"
>
  <div>
    <ul
      className="video-list"
    >
      <li
        className="video-list-item"
        onClick={[Function]}
      >
        <a
          className="video-link"
          href="/watch/cBVGlBWQzuc"
          onClick={[Function]}
        >
          <div
            className="thumbnail-wrap"
          >
            <img
              alt=""
              className="thumbnail-img"
              src="https://i.ytimg.com/vi/cBVGlBWQzuc/mqdefault.jpg"
            />
          </div>
          <p
            className="video-title"
          >
            Maroon 5 - Girls Like You ft. Cardi B (Volume 2)
          </p>
          <p
            className="channel-title"
          >
            Maroon5VEVO
          </p>
          <p
            className="video-description"
          >
            1M
             views - 
            a day ago
          </p>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
